### PR TITLE
wasm: fix double sent local reply on VM failure.

### PR DESCRIPTION
**Pull request by mathetake**
Previously the sendLocalReply was invoked twice on the VM failure, so this PR fixes that.

Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

**Comment by PiotrSikora** [comment]
> Tests and description will follow soon.

Any update? Or is this blocked or superseded by another PR?

**Comment by mathetake** [comment]
@PiotrSikora sorry I just added tests now. PTAL!

**Review by PiotrSikora — APPROVED** [review]
LGTM, thanks!

File: source/extensions/common/wasm/context.h [hunk 1]
```diff
@@ -446,11 +446,12 @@ class Context : public proxy_wasm::ContextBase,
   const Http::ResponseTrailerMap* access_log_response_trailers_{};
 
   // Temporary state.
-  ProtobufWkt::Struct temporary_metadata_;
-  bool end_of_stream_;
+  Buffer buffer_;
   bool buffering_request_body_ = false;
   bool buffering_response_body_ = false;
-  Buffer buffer_;
+  bool end_of_stream_;
```
Comment (lizan) [inline source/extensions/common/wasm/context.h:452]: init with false?

Comment (mathetake) [reply source/extensions/common/wasm/context.h]: https://github.com/envoyproxy/envoy/pull/16593/commits/d78b36eefdd0eefcdc89eb39cfd23519dd540cb2 done!

**Review by PiotrSikora — APPROVED** [review]

**Pull request merged**